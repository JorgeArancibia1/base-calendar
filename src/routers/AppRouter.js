import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { startChecking } from '../redux/actions/auth';
import { useSelector } from 'react-redux';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {
	const dispatch = useDispatch();
	const { checking, uid } = useSelector((state) => state.auth);

	useEffect(() => {
		dispatch(startChecking());
	}, [dispatch]);

	if (checking) {
		return <h1>ESPERE...</h1>;
	}

	return (
		<Router>
			<Switch>
				<PublicRoute
					exact
					path='/login'
					component={LoginScreen}
					isAuthenticated={!!uid}
				/>

				<PrivateRoute exact path='/' component={CalendarScreen} isAuthenticated={!!uid} />
				<Redirect to='/' />
			</Switch>
		</Router>
	);
};
