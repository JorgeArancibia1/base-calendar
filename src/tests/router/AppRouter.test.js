import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { AppRouter } from '../../routers/AppRouter';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)


// store.dispatch = jest.fn() // con esto se puede saber que argumentos o que fue lo que se utilizo para llamar el dispatch

describe('Pruevas en el <AppRouter />', () => {
  // Dentro de las pruebas se puede hacer match con el snapshoot que se muestre
  test('Debe mostrar el espere', () => {

    const initState = {
      auth: {
        checking: true
      }
    }

    const store = mockStore(initState)

    const wrapper = mount(
      <Provider store={store} >
        <AppRouter />
      </Provider>
    )
    // 2 Formas de muchas:
    // Con Snapshot
    expect(wrapper).toMatchSnapshot()
    // Sin Snapshot
    expect(wrapper.find('h1').exists()).toBe(true) // Que busque en el wrapper si existe una etiqueta h5.
  })
})