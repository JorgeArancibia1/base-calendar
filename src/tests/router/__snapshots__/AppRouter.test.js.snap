// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en el <AppRouter /> Debe mostrar el espere 1`] = `
<h1>
  ESPERE...
</h1>
`;
